test/BlockchainParser.test.ts

```diff
@@ -22,15 +23,38 @@ describe("Test BlockchainParser", () => {
         });
     });
 
+    describe("Test getBlocksToParse", () => {
+        const getBlocksToParse = new BlockchainParser().getBlocksToParse;
+
+        const tests = [
+            {startBlock: 2, endBlock: 2, maxConcurrentBlocks: 1, expected: 1},
+            {startBlock: 1, endBlock: 2, maxConcurrentBlocks: 1, expected: 1},
+            {startBlock: 1, endBlock: 2, maxConcurrentBlocks: 2, expected: 2},
+            {startBlock: 1, endBlock: 2, maxConcurrentBlocks: 10, expected: 2},
+            {startBlock: 1, endBlock: 3, maxConcurrentBlocks: 1, expected: 1},
+            {startBlock: 1, endBlock: 3, maxConcurrentBlocks: 2, expected: 2},
+            {startBlock: 1, endBlock: 3, maxConcurrentBlocks: 3, expected: 3},
+            {startBlock: 1, endBlock: 3, maxConcurrentBlocks: 4, expected: 3},
```

Comment:
did you try testing ascending and descending logic as well?

Reply:
Yes